packages/desktop-client/src/gocardless.ts

```diff
@@ -1,33 +1,44 @@
+import type { pushModal as pushModalAction } from 'loot-core/src/client/actions/modals';
 import { send } from 'loot-core/src/platform/client/fetch';
 
-function _authorize(pushModal, upgradingAccountId, { onSuccess, onClose }) {
+function _authorize(
+  pushModal: typeof pushModalAction,
+  upgradingAccountId: string,
+  {
+    onSuccess,
+    onClose,
+  }: {
+    onSuccess: (data: { id: string; accounts: unknown[] }) => Promise<void>;
+    onClose?: () => void;
+  },
+) {
   pushModal('gocardless-external-msg', {
     onMoveExternal: async ({ institutionId }) => {
-      const accessValidForDays = 30;
       const resp = await send('gocardless-create-web-token', {
         upgradingAccountId,
         institutionId,
-        accessValidForDays,
+        accessValidForDays: 30,
       });
 
-      if (resp.error) return resp;
+      if ('error' in resp) return resp;
       const { link, requisitionId } = resp;
       window.Actual.openURLInBrowser(link);
 
-      let { error, data } = await send('gocardless-poll-web-token', {
+      return send('gocardless-poll-web-token', {
         upgradingAccountId,
         requisitionId,
       });
-
-      return { error, data };
     },
 
     onClose,
     onSuccess,
   });
 }
 
-export async function authorizeBank(pushModal, { upgradingAccountId } = {}) {
+export async function authorizeBank(
+  pushModal: typeof pushModalAction,
+  { upgradingAccountId }: { upgradingAccountId?: string } = {},
```

Comment:
Should we extract this to a type?

Reply:
What benefit would that give? It's not really a complex type that we're reusing elsewhere. Just a primitive `string`

Reply:
Just to be consistent with the other parts of the codebase but don't consider it as a blocker :)